lib/beaker/hypervisor/vcloud.rb

```diff
@@ -73,16 +73,22 @@ def enable_root(host)
 
     def create_clone_spec(host)
       # Add VM annotation
+      additionalExtraConfig = if host['extraConfig']
+                                host['extraConfig'].map do |key, value|
+                                  { key: key, value: value }
+                                end
+                              else
+                                []
+                              end
+      extraConfig = [{ key: 'guestinfo.hostname', value: host['vmhostname'] }] + additionalExtraConfig
```

Comment:
Is this easier to read?
```suggestion
      extraConfig = [{ key: 'guestinfo.hostname', value: host['vmhostname'] }] 
      if host['extraConfig']
        extraConfig += host['extraConfig'].map do |key, value|
          { key: key, value: value }
        end
      end
```

I also wonder what happens if `guestinfo.hostname` is in `host['extraConfig']`.

Makes me tempted to write:
```ruby
extraConfig = { 'guestinfo.hostname' => host['vmhostname'] }
extraConfig.merge(host['extraConfig']) if host['extraConfig']
```

And later:
```ruby
extraConfig: extraConfig.map { |k, v| { key: k, value: v } }
```